src/InstrumentationEngine.Lib/InstrumentationMethod.cpp

```diff
@@ -83,7 +83,17 @@ HRESULT MicrosoftInstrumentationEngine::CInstrumentationMethod::InitializeCore(
     WCHAR wszModuleFullPath[MAX_PATH];
     memset(wszModuleFullPath, 0, MAX_PATH);
     wcscpy_s(wszModuleFullPath, MAX_PATH, m_bstrModuleFolder);
-    PathCchAppend(wszModuleFullPath, MAX_PATH, m_bstrModule);
+
+    // For Windows 7 support, we cannot use PathCch functions.
+    //PathCchAppend(wszModuleFullPath, MAX_PATH, m_bstrModule);
+    if (!wcslen(wszModuleFullPath) + m_bstrModule.Length() < MAX_PATH - 1 ||
```

Comment:
I assume you are doing `MAX_PATH - 1` to allow the null terminator. I don't think this accounts for the directory separators?

Reply:
Correct about the null terminator, and there's a slight problem with PatchCanonicalize which can cause paths to be longer so I'll need to refactor this with more thought